### PR TITLE
Social thread

### PR DESCRIPTION
**Pull request by zacheryconverse**
# Submit a pull request

### 🎯 Goal

Build thread component

### 🛠 Implementation details

Add message options to access thread with header and inner

### 🎨 UI Changes

Screen Shot 2021-09-24 at 11 15 11 AM

### CODE REVIEW

File: examples/website-demos/social-messaging/src/assets/GiphyIcon.tsx [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 
 export const GiphyIcon: React.FC = () => (
-  <svg width='70' height='24' viewBox='0 0 70 24' fill='none' xmlns='http://www.w3.org/2000/svg'>
+  <svg
+    width='70'
+    height='24'
+    viewBox='0 0 70 24'
+    fill='none'
+    xmlns='http://www.w3.org/2000/svg'
+    style={{ flexShrink: 0 }}
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/assets/GiphyIcon.tsx:10]: why is this style needed?

File: examples/website-demos/social-messaging/src/components/ChannelSearch/SearchResultItem.tsx [hunk 2]
```diff
@@ -12,12 +12,12 @@ import {
...
-  SocialCommandType,
-  SocialEventType,
-  SocialMessageType,
-  SocialReactionType,
+  // SocialAttachmentType,
+  // SocialChannelType,
+  // SocialCommandType,
+  // SocialEventType,
+  // SocialMessageType,
+  // SocialReactionType,
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/ChannelSearch/SearchResultItem.tsx:20]: are we not using the generics in this component? if not we can delete

File: examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx [hunk 3]
```diff
@@ -64,10 +64,10 @@ export const ChatContainer: React.FC = () => {
   const [chatClient, setChatClient] = useState<StreamChat>();
 
   const {
-    chatsUnreadCount,
+    // chatsUnreadCount,
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx:67]: do we need to keep these commented out values in the file?

File: examples/website-demos/social-messaging/src/components/ChannelContainer/ChannelContainer.tsx [hunk 4]
```diff
@@ -81,7 +82,7 @@ export const ChannelContainer: React.FC = () => {
         {isNewChat ? <NewChat /> : <MessageList />}
         <MessageInput mentionAllAppUsers />
       </Window>
-      {/* <Thread /> */}
+      <SocialThreadInner />
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/ChannelContainer/ChannelContainer.tsx:85]: let's just call this SocialThread, an Inner component would be one nested as a child within the outer layer

File: examples/website-demos/social-messaging/src/components/Thread/SocialThread.scss [hunk 5]
```diff
@@ -1,67 +1,67 @@
...
-  margin-top: auto;
-}
-
-.str-chat__thread .str-chat__message-replies-count-button {
-  display: none;
+.str-chat__thread {
+  margin: 0 !important;
+  border-left: 1px solid $stroke !important;
+  flex: 1 !important;
+  max-width: none !important;
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/Thread/SocialThread.scss:57]: nest this selector to try and get rid of the importants, this comment goes for all importants in the PR

File: examples/website-demos/social-messaging/src/components/Thread/SocialThreadInner.tsx [hunk 6]
```diff
@@ -21,6 +21,7 @@ export const ThreadInner = () => {
   return (
     <>
       <Thread
+        // fullWidth
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/Thread/SocialThreadInner.tsx:24]: can this be deleted?

File: examples/website-demos/social-messaging/src/components/Thread/SocialThreadHeader.tsx [hunk 7]
```diff
@@ -20,9 +22,6 @@ export const SocialThreadHeader: React.FC = () => {
           with {thread.user?.name || thread.user?.id || ''}
         </div>
       </div>
-      <div className='thread-header-actions'>
-        <Ellipse />
-      </div>
```
Comment (DanC5) [inline examples/website-demos/social-messaging/src/components/Thread/SocialThreadHeader.tsx:25]: don't need the Ellipse anymore?

File: examples/website-demos/social-messaging/src/assets/GiphyIcon.tsx (hunk 1 above)
Comment (zacheryconverse) [reply examples/website-demos/social-messaging/src/assets/GiphyIcon.tsx:10]: When the channelContainer shrinks to accommodate the thread, the GiphyIcon shrinks as well without this style

File: examples/website-demos/social-messaging/src/components/ChannelSearch/SearchResultItem.tsx (hunk 2 above)
Comment (zacheryconverse) [reply examples/website-demos/social-messaging/src/components/ChannelSearch/SearchResultItem.tsx:20]: I just commented these out in this unrelated/unfinished component to limit console warnings

File: examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx (hunk 3 above)
Comment (zacheryconverse) [reply examples/website-demos/social-messaging/src/components/ChatContainer/ChatContainer.tsx:67]: same as above comment

File: examples/website-demos/social-messaging/src/components/Thread/SocialThreadHeader.tsx (hunk 7 above)
Comment (zacheryconverse) [reply examples/website-demos/social-messaging/src/components/Thread/SocialThreadHeader.tsx:25]: Not according to the design

File: examples/website-demos/social-messaging/src/components/Thread/SocialThread.scss (hunk 5 above)
Comment (zacheryconverse) [reply examples/website-demos/social-messaging/src/components/Thread/SocialThread.scss:57]: on it

**Review by DanC5 — APPROVED** [review]

**Pull request merged**